lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -287,7 +286,22 @@ func (handler *StorageHandlersImpl) CreateVolume(params storage.CreateVolumePara
 
 	volume, err := storageVolumeLayer.VolumeCreate(context.TODO(), params.VolumeRequest.Name, storeURL, capacity*1024, byteMap)
 	if err != nil {
-		log.Errorf("storagehandler: VolumeCreate error: %s", err)
+		log.Errorf("storagehandler: VolumeCreate error: %#v", err)
+
+		if os.IsExist(err) {
+			return storage.NewCreateVolumeConflict().WithPayload(&models.Error{
+				Code:    swag.Int64(http.StatusNotFound),
```

Comment:
`NewVolumeConflict()` would set a `409` on the http response, yet you have `404` set in the body of the error code?